Corrections.py

```diff
@@ -158,18 +158,24 @@ def getNormalisationCorrections(df, config, sample, nLegs, ana_cache=None, retur
     df, pu_SF_branches = pu.getWeight(df)
     df = df.Define('genWeightD', 'std::copysign<double>(1., genWeight)')
     all_branches = [ pu_SF_branches ]
+    #print(pu_SF_branches)
     all_sources = set(itertools.chain.from_iterable(all_branches))
     all_sources.remove(central)
     all_weights = []
     denom = f'/{ana_cache["denominator"][central][central]}' if ana_cache is not None else ''
-
+    #print(ana_cache['denominator'].keys())
+    #print(f"all_sources = {list(all_sources)}")
     for syst_name in [central] + list(all_sources):
         #if not isCentral : continue
         branches = getBranches(syst_name, all_branches)
         product = ' * '.join(branches)
         weight_name = f'weight_{syst_name}' if syst_name!=central else 'weight_total'
         weight_rel_name = weight_name + '_rel'
         weight_out_name = weight_name if syst_name == central else weight_rel_name
+        for scale in ['Up', 'Down']:
+            if syst_name == f'pu{scale}':
+                #print(f"using anacache[denom][pu][{scale}]")
+                denom = f"""/{ana_cache["denominator"]["pu"][scale]}""" if ana_cache is not None else ''
```

Comment:
Like this, it changes `denom` also for the consecutive executions. `denom` definition should be done inside `for`